indico/modules/events/timetable/views/weeks.py

```diff
@@ -105,5 +106,14 @@ def inject_week_timetable(event, days, tz_name):
             tmp.append((day, day_entries))
         week_table.append(tmp)
 
-    return render_template('events/timetable/display/_weeks.html', event=event, week_table=week_table,
+    entry_colors = {}
+    if (layout_settings.get(event, 'weeks_view_settings') and
+            'colors' in layout_settings.get(event, 'weeks_view_settings')):
+        entry_colors = layout_settings.get(event, 'weeks_view_settings')['colors']
+    elif (theme_settings.themes['indico-weeks-view'].get('arguments') and
+          theme_settings.themes['indico-weeks-view'].get('arguments').get('colors') and
+          theme_settings.themes['indico-weeks-view'].get('arguments').get('colors').get('default')):
```

Comment:
You could just do this and get rid of the previous two lines...

    theme_settings.themes['indico-weeks-view'].get('arguments', {}).get('colors', {}).get('default')